File: chordes.cpp
#include <iostream>
#include <sstream>

using namespace std;

string getChordURL(string chord) {
	string res = "https://tuneronline.ru/images/chords/guitar/max/";
	for (int i = 0; i < chord.size(); i++) {
		chord[i] = tolower(chord[i]);
		if (chord[i] == '#') res += "diez";
		else res.push_back(chord[i]);
	}
	res += ".jpg";
	return res;
}
const int IMG_SIZE = 50;

int main() {
	string word;
	while (getline(cin, word)) {
		stringstream ss;
		ss << word;
		
		while (ss >> word) {
			if (true) { // chord
				bool brackets = (word.front() == '(' && word.back() == ')');
				if (brackets) word = word.substr(1, word.size() - 2);
				
				if (brackets) cout << '(';
				cout << "<img src=\"" << getChordURL(word) << "\" height=\"" << IMG_SIZE << "\">" << endl;
				
				if (brackets) cout << ')';
			}
		}
		
		cout << "<br>" << endl;
	}
	
	return 0;
}

File: chordes.py
import sys

NOTES = "abcdefgh"


def get_chord_url(chord):
	res = ''
	for char in chord.lower():
		if char in ('#', '♯'):
			res += 'diez'
		elif char in ('b', '♭'):
			res += 'b'
		else:
			res += char
	return f'https://tuneronline.ru/images/chords/guitar/max/{res}.jpg'


def big_font(x):
	return f'<font face="sans-serif" size="5">{x}</font>'


if len(sys.argv) <= 2:
	print('usage: song.txt song.html')
	exit(0)

fin = open(sys.argv[1], 'r')
fout = open(sys.argv[2], 'w')

print('<meta charset="utf-8">', file=fout)

print('<table>', file=fout)
for line in fin:
	print('<tr>\n<td>', file=fout)
	
	for word in line.split():
		brackets = word[0] == '(' and word[-1] == ')'
		if (brackets and word[1].lower() not in NOTES) or (not brackets and word[0].lower() not in NOTES):
			print(big_font(word), file=fout)
	
	print('</td>\n<td>', file=fout)
	
	for word in line.split():
		brackets = word[0] == '(' and word[-1] == ')'
		if brackets:
			word = word[1:-1]
			print(big_font('('), file=fout)
		
		if word[0].lower() in NOTES:
			print(f'<img src="{get_chord_url(word)}" height="90">', file=fout)
		
		if brackets:
			print(big_font(')'), file=fout)
	
	print('</td>\n</tr>', file=fout)
print('</table>', file=fout)

print('</font>', file=fout)

fin.close()
fout.close()

print(f'Check result: {sys.argv[2]}!')
